wordpress-mobile/WordPress-Android
# [Blogging Prompts] Prompts List architectural boilerplate

Part of #17125

Add architectural boilerplate for fetching data and propagating UI state to the Compose screen by creating ViewModel, UseCase, UI State, and view objects, and using fake data for now.

I also created a couple of placeholder UIs for the Content and Loading states, so the changes could be viewed / tested from a usability perspective as well. Note: the UI seen here is not close to the actual spec, that's why there are a couple of TODO comments left in code.

<details>
<summary>Video Demo</summary>

https://user-images.githubusercontent.com/5091503/209221284-66181422-ab80-4518-8780-1cf79d51449f.mp4

Note: the UI is not the final spec and the data is fake at this point.
</details>

To test:
1. Do the steps to enable the Prompts List feature flag found in the test section of https://github.com/wordpress-mobile/WordPress-Android/pull/17675
2. Sign in (if not signed in already)
3. Go to the home dashboard (My Site -> Home)
4. **Verify** the Prompts card is shown
5. Tap the 3-dot menu in the Prompts card
6. Tap the "View more prompts" option
7. **Verify** the Prompts List placeholder activity is shown
8. **Verify** it shows a "Loading..." text for a while
9. **Verify** the content is updated with a fake list of 11 prompts

## Regression Notes
1. Potential unintended areas of impact
- Analytics tracking for the screen view not triggering

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual tests

3. What automated tests I added (or what prevented me from doing so)
- ViewModel tests
- UseCase test (just a simple one since fake data generation is happening here for now)

PR submission checklist:
